sdcm/utils/benchmarks.py

```diff
@@ -0,0 +1,137 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2021 ScyllaDB
+import logging
+import re
+
+from sdcm.remote import RemoteCmdRunnerBase, shell_script_cmd
+from sdcm.utils.metaclasses import Singleton
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ScyllaClusterBenchmarkManager(metaclass=Singleton):
+    def __init__(self):
+        """
+        This class is meant for gathering the benchmark results
+        of all the relevant nodes in the cluster and presenting
+        them in unified fashion.
+        Optionally, these can also be committed to remote
+        storage (S3? Argus?)
+        """
+        self._nodes: list["BaseNode"] = []
+        self._benchmark_runners: list[ScyllaNodeBenchmarkRunner] = []
+
+    def add_node(self, new_node: "BaseNode"):
+        if new_node.distro.is_debian_like:
+            self._benchmark_runners.append(ScyllaNodeBenchmarkRunner(new_node))
+        else:
+            LOGGER.info("Skipped installing benchmarking tools on a non-debian-like distro.")
+
+    def install_benchmark_tools(self):
+        for runner in self._benchmark_runners:
+            runner.install_benchmark_tools()
+
+    def run_benchmarks(self):
+        for runner in self._benchmark_runners:
+            runner.run_benchmarks()
+
+    @staticmethod
+    def _collect_benchmark_output():
```

Comment:
It looks like as "abstract" method which must be redefined. So, I guess, it should have "@abc.abstractmethod" decorator.

Reply:
Fixed with rebase